skyportal/handlers/api/allocation.py

```diff
@@ -0,0 +1,182 @@
+from ..base import BaseHandler
+from ...models import DBSession, Group, Allocation, Instrument
+from baselayer.app.access import auth_or_token, permissions
+from marshmallow.exceptions import ValidationError
+
+
+class AllocationHandler(BaseHandler):
+    @auth_or_token
+    def get(self, allocation_id=None):
+        """
+        ---
+        single:
+          description: Retrieve an allocation
+          parameters:
+            - in: path
+              name: allocation_id
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+               content:
+                application/json:
+                  schema: SingleAllocation
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all allocations
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfAllocations
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+
+        # get owned allocations
+        allocations = (
+            DBSession()
+            .query(Allocation)
+            .filter(
+                Allocation.group_id.in_(
+                    [g.id for g in self.current_user.accessible_groups]
+                )
+            )
+        )
+
+        if allocation_id is not None:
```

Comment:
Why not just have a separate query for this case? It'd simplify some of the logic below, specifically everywhere we check whether `allocation_id is not None`, and those lines can just be moved inside this block. This block could (should) then also be moved to the top with its own return statement, then everything following would be for the "multiple" case. That's a pattern we use in several other GET handlers. Trying to make a single query generalized for all cases seems to actually increase complexity, but I imagine the intent is the opposite

Reply:
done